### PR TITLE
Sama-COCO Dataset Integration

### PR DESCRIPTION
**Pull request by EricZimmermann**
## What changes are proposed in this pull request?
Addition of Sama-COCO dataset
- Sama-COCO utils
- Sama-COCO dataset

## How is this patch tested? If it is not, please explain why.

Local validation by loading zoo dataset (train and validation detections and masks)

### Is this a user-facing change that should be mentioned in the release notes?

-   [ ] No. You can skip the rest of this section.
-   [x] Yes. Give a description of this change to be included in the release
        notes for FiftyOne users.

The addition of Sama-COCO to the zoo datasets

### What areas of FiftyOne does this PR affect?

-   [ ] App: FiftyOne application changes
-   [ ] Build: Build and test infrastructure changes
-   [ ] Core: Core `fiftyone` Python library changes
-   [ ] Documentation: FiftyOne documentation changes
-   [x] Other

### CODE REVIEW

File: fiftyone/zoo/datasets/base.py [hunk 1]
```diff
@@ -1173,6 +1174,206 @@ def _get_raw_dir(self, dataset_dir):
...
+    -   Validation split: 5,000 images
+
+    Notes:
+
+    -   COCO defines 91 classes but the data only uses 80 classes
+    -   Some images from the train and validation sets don't have annotations
+    -   The test set does not have annotations
+    -   Sama-COCO may have some discrepancies with COCO-2017 in terms of the instances labeled
+
+    #TODO: update example usage
```
Comment (ehofesmann) [inline fiftyone/zoo/datasets/base.py:1207]: Update example to use sama-coco instead

File: docs/source/user_guide/dataset_zoo/datasets.rst [hunk 2]
```diff
@@ -3391,6 +3393,233 @@ annotation data.
...
+        #
+
+        fiftyone zoo datasets load sama-coco \
+            --split validation \
+            --kwargs \
+                label_types=detections,segmentations
+
+        fiftyone app launch sama-coco-validation
+
+.. image:: /images/dataset_zoo/sama-coco-validation.png
```
Comment (ehofesmann) [inline docs/source/user_guide/dataset_zoo/datasets.rst:3619]: Include this image in the PR

**Review by ehofesmann — CHANGES_REQUESTED** [review]
Thanks for the PR Eric! (Great name btw :wink: )

This is looking really good, it works great. Just left two small comments for you after which I believe it should be good to merge!

File: docs/source/user_guide/dataset_zoo/datasets.rst (hunk 2 above)
Comment (EricZimmermann) [reply docs/source/user_guide/dataset_zoo/datasets.rst:3619]: Image added - any requirements on aspect ratios?

File: fiftyone/zoo/datasets/base.py (hunk 1 above)
Comment (EricZimmermann) [reply fiftyone/zoo/datasets/base.py]: All refs to original coco have been resolved (dataset source, naming, examples)

File: docs/source/user_guide/dataset_zoo/datasets.rst (hunk 2 above)
Comment (ehofesmann) [reply docs/source/user_guide/dataset_zoo/datasets.rst:3619]: Thanks! No hard requirements on aspect ratios, the image you added looks great!

**Review by ehofesmann — APPROVED** [review]
LGTM!

**Pull request merged**